### DISCUSSION TITLE
Request for approving three ACT Rules

### DISCUSSION THREAD

**Discussion by WilcoFiers** [w3c/aria#2078] [Show and tell]
The ACT Task Force has completed work on the following three rules, and request approval from the ARIA working group to publish these on the W3C website. If you have comments, please respond to the thread. This is a follow-up on our previous [discussion](https://github.com/w3c/aria/discussions/1850). I believe these three rules are ready. Others may take a bit longer, particularly the required context / required owned element rules as there are various changes in ARIA 1.3 to work through first.

If you are not familiar with ACT Rules, please first read [about ACT Rules](https://www.w3.org/WAI/standards-guidelines/act/rules/about/) and the [ACT Rules format 1.0](https://www.w3.org/TR/act-rules-format/)

Please vote 👍 if you approve of the change, or leave a comment if you have concerns.

**Comment by WilcoFiers** [comment]
## ARIA attribute is defined in WAI-ARIA

https://www.w3.org/WAI/standards-guidelines/act/rules/5f99a7/proposed/

This rule checks that each aria- attribute specified is defined in ARIA 1.2.

**Reply by rahimabdi** [reply]
Hi @WilcoFiers, unsure if this is implementation-specific: does this rule validate `aria-labelledby` and `aria-labeledby` as synonyms? Please see https://github.com/w3c/aria/issues/2093.

**Reply by WilcoFiers** [reply]
🤯 Nope, it sure doesn't. ARIA 1.2 doesn't mention `aria-labeledby` as a synonym. It would have to be added to the standard first I think. I just did a quick test of it, and it doesn't yet seem to work in Firefox. We're following the spec here, so if ARIA 1.3 adds `aria-labeledby` as a synonym then the rule would allow it.

**Comment by WilcoFiers** [comment]
## ARIA state or property has valid value

https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/proposed/

This rule checks that each ARIA state or property has a valid value type.

**Reply by JAWS-test** [reply]
> Passed Example 3: The aria-expanded [attribute value](https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/proposed/#attribute-value) of undefined is allowed for the true/false/undefined [value type](https://www.w3.org/TR/wai-aria-1.2/#propcharacteristic_value)

I'm not sure if the ARIA specification allows for the use of undefined. My understanding is that undefined is always given if the attribute does not occur. That's why tristate is also called tristate, because it only has the three valid states true, false and mixed.
For example, a checkbox cannot be aria-checked=undefined, whereas all elements that may not have aria-checked are undefined with regard to aria-checked.

See: https://github.com/w3c/aria/issues/2079

**Reply by WilcoFiers** [reply]
Thanks for the feedback. It does seem like it's currently allowed, but I appreciate that this might not be an intended use. I would propose we keep this as is for now, and if ARIA ends up prohibiting the use of this value later we can update the rule to reflect that.

**Comment by WilcoFiers** [comment]
## Role attribute has valid value

https://www.w3.org/WAI/standards-guidelines/act/rules/674b10/proposed/

This rule checks that each role attribute has a valid value.

**Reply by rahimabdi** [reply]
And also, an FYI on this issue related to `role="text"` support: https://github.com/w3c/aria/issues/2011.

**Reply by WilcoFiers** [reply]
Thanks for sharing. Since the rule doesn't allow `role=text` that shouldn't make a difference here. I'll share it with other tool devs though. I don't know which of them allow `role=text`, but it might change some people's minds about allowing it.

**Comment by WilcoFiers** [comment]
As confirmed via e-mail by @jnurthen and @spectranaut, all three rules are approved. This should happen some time today.
